core/templates/pages/admin-page/features-tab/admin-features-tab.component.ts

```diff
@@ -213,11 +222,15 @@ export class AdminFeaturesTabComponent implements OnInit {
       this.featureFlagNameToBackupMap.set(feature.name, cloneDeep(feature));
 
       this.setStatusMessage.emit('Saved successfully.');
-    } catch (e) {
-      if (e.error && e.error.error) {
-        this.setStatusMessage.emit(`Update failed: ${e.error.error}`);
-      } else {
-        this.setStatusMessage.emit('Update failed.');
+    // The type of error 'e' is unknown because it could be anything.
+    // This gave the freedom to access any property of 'e' at any time.
+    } catch (e: unknown) {
+      if (e instanceof HttpErrorResponse) {
```

Comment:
And what if it isn't? (In general, can this case occur? Should an exception be thrown)

Reply:
Yes surprisingly, this case is not happening. It always stands on if statement.

Reply:
FYI, I checked with @vojtechjelinek about this. It seems incorrect to use if/else here because the else case is not being taken care of properly -- if it can happen, we should do something with it, and if it can't happen, we should do a type assertion. (I think he's just posted a review -- see his other comments.)

Reply:
https://github.com/oppia/oppia/pull/15089#discussion_r839740119

Reply:
When assert is used, it will work by throwing an error when it encounters an unexpected scenario, but it will still execute code after the error is thrown.
![Screenshot from 2022-04-01 13-48-36](https://user-images.githubusercontent.com/80846488/161368316-a68a9d3d-5034-41e5-b630-d8156ecf2fc6.png)
Also check is it right way to do, actually never use assert before in JS/TS. Need review.

Reply:
I'm not sure about this solution, but I read through the docs and I am wondering about whether we should just add an "else throw" clause to what was there before. That way, if an unexpected error occurs, we can learn about it.

I suppose if you use asserts you will need a "throw new AssertionError" clause somewhere so that an error actually gets thrown? (Like in the examples in the docs that @vojtechjelinek linked to)

Reply:
Already tried no difference, same reason.

Reply:
I don't understand what you've tried. Where is the "else throw"?

Reply:
@seanlip i try using assert in every way, but it still execute code after the error is thrown and unable to catch `e.error`.
Please see current way. Is it Okay?

Reply:
Current way looks fine to me, thanks. @vojtechjelinek thoughts?

Reply:
Looks good.